benchmarks/diffusion_2D/main_arkode.cpp

```diff
@@ -205,6 +217,50 @@ int main(int argc, char* argv[])
         if (check_flag(&flag, "SUNLinSolSetInfoFile_SPGMR", 1)) return(1);
       }
     }
+    else
+    {
+#if defined(USE_SUPERLU_DIST)
+      // initialize SuperLU-DIST grid
+      superlu_gridinit(udata.comm_c, udata.npx, udata.npy, &grid);
+
+      // Initialize sparse matrix data structure and SuperLU_DIST solver
+      sunindextype nnz_loc = 5 * udata.nodes_loc;
+
+      A_data = (realtype*)malloc(nnz_loc * sizeof(sunrealtype));
+      if (check_flag((void*)A_data, "malloc Adata", 0)) return 1;
+
+      A_col_idxs = (sunindextype*)malloc(nnz_loc * sizeof(sunindextype));
+      if (check_flag((void*)A_col_idxs, "malloc Acolind", 0)) return 1;
+
+      A_row_ptrs = (sunindextype*)malloc((udata.nodes_loc + 1) * sizeof(sunindextype));
+      if (check_flag((void*)A_row_ptrs, "malloc Arowptr", 0)) return 1;
+
+      // SuperLU_DIST structures
+      dCreate_CompRowLoc_Matrix_dist(&A_super, udata.nodes, udata.nodes,
+                                     nnz_loc, udata.nodes_loc, 0, A_data,
+                                     A_col_idxs, A_row_ptrs, SLU_NR_loc, SLU_D,
+                                     SLU_GE);
+      dScalePermstructInit(udata.nodes, udata.nodes, &A_scaleperm);
+      dLUstructInit(udata.nodes, &A_lu);
+      PStatInit(&A_stat);
+      set_default_options_dist(&A_opts);
+      A_opts.PrintStat = NO;
+
+      // SUNDIALS structures
+      A = SUNMatrix_SLUNRloc(&A_super, &grid, ctx);
+      if (check_flag((void*)A, "SUNMatrix_SLUNRloc", 0)) return 1;
+
+      LS = SUNLinSol_SuperLUDIST(u, A, &grid, &A_lu, &A_scaleperm, &A_solve,
+                                 &A_stat, &A_opts, ctx);
+      if (check_flag((void*)LS, "SUNLinSol_SuperLUDIST", 0)) return 1;
+
+      // Disable preconditioning
+      uopts.prec = false;
```

Comment:
If we used a variable name like `uopts.enable_preconditioning` then the code would be self-documenting and the comment would not be needed.